internal/monitor/otelexporters.go

```diff
@@ -36,20 +40,57 @@ const serviceName = "gcsfuse"
 
 // SetupOTelMetricExporters sets up the metrics exporters
 func SetupOTelMetricExporters(ctx context.Context, c *cfg.Config) (shutdownFn common.ShutdownFn) {
+	shutdownFns := make([]common.ShutdownFn, 0)
 	options := make([]metric.Option, 0)
+
 	opts, shutdownFn := setupPrometheus(c.Metrics.PrometheusPort)
 	options = append(options, opts...)
+	shutdownFns = append(shutdownFns, shutdownFn)
+
+	opts, shutdownFn = setupCloudMonitoring(c.Metrics.CloudMetricsExportIntervalSecs)
+	options = append(options, opts...)
+	shutdownFns = append(shutdownFns, shutdownFn)
+
 	res, err := getResource(ctx)
 	if err != nil {
 		logger.Errorf("Error while fetching resource: %v", err)
 	} else {
 		options = append(options, metric.WithResource(res))
 	}
+
 	meterProvider := metric.NewMeterProvider(options...)
+	shutdownFns = append(shutdownFns, meterProvider.Shutdown)
+
 	otel.SetMeterProvider(meterProvider)
-	return common.JoinShutdownFunc(meterProvider.Shutdown, shutdownFn)
+
+	return common.JoinShutdownFunc(shutdownFns...)
 }
 
+func setupCloudMonitoring(secs int64) ([]metric.Option, common.ShutdownFn) {
+	if secs <= 0 {
+		return nil, nil
+	}
+	options := []cloudmetric.Option{
+		cloudmetric.WithMetricDescriptorTypeFormatter(metricFormatter),
+		cloudmetric.WithFilteredResourceAttributes(func(kv attribute.KeyValue) bool {
+			// Ensure that PID is available as a metric label on metrics explorer.
+			return cloudmetric.DefaultResourceAttributesFilter(kv) ||
+				kv.Key == semconv.ProcessPIDKey
+		}),
+	}
+	exporter, err := cloudmetric.New(options...)
+	if err != nil {
+		logger.Errorf("Error while creating Google Cloud exporter:%v", err)
+		return nil, nil
+	}
+
+	r := metric.NewPeriodicReader(exporter, metric.WithInterval(time.Duration(secs)*time.Second))
+	return []metric.Option{metric.WithReader(r)}, r.Shutdown
+}
+
+func metricFormatter(m metricdata.Metrics) string {
+	return "custom.googleapis.com/gcsfuse/" + strings.ReplaceAll(m.Name, ".", "/")
```

Comment:
Pls move the constants to top.

Reply:
https://engdoc.corp.google.com/eng/doc/devguide/java/practices/constants.md?cl=head#when-to-use-a-constant-versus-an-inlined-literal - I think in this case an inlined literal makes more sense as opposed to a constant.

Reply:
Done